packages/eth-providers/src/__tests__/e2e/json-rpc-provider.test.ts

```diff
@@ -21,14 +21,23 @@ describe('JsonRpcProvider', async () => {
   const someOne = '0xf7ABcfa42bF7e7d43d3d53C665deD80fDAfB5244';
 
   const provider = new AcalaJsonRpcProvider('https://eth-rpc-karura.aca-api.network');
+  await provider.ready;
   const usdcAddr = '0x1F3a10587A20114EA25Ba1b388EE2dD4A337ce27';
   const usdc = new Contract(usdcAddr, erc20Json.abi, provider);
 
   /* --------- local --------- */
   const testKey = 'a872f6cbd25a0e04a08b1e21098017a9e6194d101d75e13111f71410c59cd57f';   // 0x75E480dB528101a381Ce68544611C169Ad7EB342
   const providerLocal = new AcalaJsonRpcProvider(localEthRpc);
+  await providerLocal.ready;
```

Comment:
for JsonRpcProvider I think we don't need to wait, since all communication is through json rpc